lte/gateway/c/core/oai/lib/n11/SmfServiceClient.cpp

```diff
@@ -170,4 +178,76 @@ AsyncSmfServiceClient& AsyncSmfServiceClient::getInstance() {
   return instance;
 }
 
+static void handle_subs_update_location_info_ans(
```

Comment:
why do we add these to SmfServiceClient? We should use the AMF client for the subscriberdb.

Reply:
in fact strike this. You can expand the itti message for s6a and grpc handler `s6a_handle_update_location_ans` such that you can specify (in the itti request) the task to return the response to (i.e., either MME task or AMF task) and this parameter can be passed onto the handler.

Reply:
> why do we add these to SmfServiceClient? We should use the AMF client for the subscriberdb.

Right now authentication service client connects to subscriberdb, i will put the changes in this client interface

Reply:
> in fact strike this. You can expand the itti message for s6a and grpc handler `s6a_handle_update_location_ans` such that you can specify (in the itti request) the task to return the response to (i.e., either MME task or AMF task) and this parameter can be passed onto the handler.

Fixed this comment by passing the answer object onto the MME or AMF thread based on RAT type thus avoiding the redundancy of the code